pkgs/tools/misc/grub/2.0x.nix

```diff
@@ -78,6 +78,68 @@ stdenv.mkDerivation rec {
       url = "https://git.savannah.gnu.org/cgit/grub.git/patch/?id=000b5cd04fd228f9741f5dca0491636bc0b89eb8";
       sha256 = "sha256-Mex3qQ0lW7ZCv7ZI7MSSqbylJXZ5RTbR4Pv1+CJ0ciM=";
     })
+    (fetchpatch {
+      name = "CVE-2022-2601.CVE-2022-3775.1.patch";
+      url = "https://git.savannah.gnu.org/cgit/grub.git/patch/?id=f6b6236077f059e64ee315f2d7acb8fa4eda87c5";
+      sha256 = "sha256-pk02iVf/u6CdsVjl8HaFBh0Bt473ZQzz5zBp9SoBLtE=";
+    })
+    (fetchpatch {
+      name = "CVE-2022-2601.CVE-2022-3775.2.patch";
+      url = "https://git.savannah.gnu.org/cgit/grub.git/patch/?id=9c76ec09ae08155df27cd237eaea150b4f02f532";
+      sha256 = "sha256-axbEOH5WFkUroGna2XY1f2kq7+B1Cs6LiubIA2EBdiM=";
+    })
+    (fetchpatch {
+      name = "CVE-2022-2601.CVE-2022-3775.3.patch";
+      url = "https://git.savannah.gnu.org/cgit/grub.git/patch/?id=768e1ef2fc159f6e14e7246e4be09363708ac39e";
+      sha256 = "sha256-aKDUVS/Yx1c87NCrt4EG8BlSpkHijUyAJIwbmtzNjD8=";
+    })
+    (fetchpatch {
+      name = "CVE-2022-2601.CVE-2022-3775.4.patch";
+      url = "https://git.savannah.gnu.org/cgit/grub.git/patch/?id=c51292274ded3259eb04c2f1c8d253ffbdb5216a";
+      sha256 = "sha256-OLNOKuAJuHy2MBMnU2xcYM7AaxmDk9fchXhggoDrxJU=";
+    })
+    (fetchpatch {
+      name = "CVE-2022-2601.CVE-2022-3775.5.patch";
+      url = "https://git.savannah.gnu.org/cgit/grub.git/patch/?id=23843fe8947e4da955a05ad3d1858725bfcb56c8";
+      sha256 = "sha256-ptn00nqVJlEb1c6HhoMy9nrBuctH077LM4yXKsK47gc=";
+    })
+    (fetchpatch {
+      name = "CVE-2022-2601.CVE-2022-3775.6.patch";
+      url = "https://git.savannah.gnu.org/cgit/grub.git/patch/?id=b9396daf1c2e3cdc0a1e69b056852e0769fb24de";
+      sha256 = "sha256-K7XNneDZjLpZh/C908+5uYsB/0oIdgQqmk0yJrdQLG4=";
+    })
+    (fetchpatch {
+      name = "CVE-2022-2601.CVE-2022-3775.7.patch";
+      url = "https://git.savannah.gnu.org/cgit/grub.git/patch/?id=1d2015598cc7a9fca4b39186273e3519a88e80c7";
+      sha256 = "sha256-s4pZtszH4b/0u85rpzVapZmNQdYEq/wW06SQ3PW/1aU=";
+    })
+    # Patch 8 should be https://git.savannah.gnu.org/gitweb/?p=grub.git;a=commit;h=93a786a00163e50c29f0394df198518617e1c9a5
```

Comment:
Looking at the patch, is this not something that could apply on grub2_efi? Why would we have removed entirely the shim logic from our source tree...?
I'm probably missing some context here.

Reply:
In fact we are missing the patch for CVE-2022-28735 which explains why it does not apply.